go/mysqlconn/conn.go

```diff
@@ -256,7 +259,9 @@ func (c *Conn) writeOKPacket(affectedRows, lastInsertID uint64, flags uint16, wa
 	if err := c.writePacket(data); err != nil {
 		return err
 	}
-	c.flush()
+	if err := c.flush(); err != nil {
```

Comment:
It's hard to reason about the correctness (and efficiency) of flush. If a write is always followed by a read, we could have the read perform the flush.
The other option would be to have the exported functions perform the flush.
Or both.